cmd/maestro/environments/framework.go

```diff
@@ -173,11 +176,28 @@ func (e *Env) LoadClients() error {
 		glog.Infof("Using Mock CloudEvents Source Client")
 		e.Clients.CloudEventsSource = cloudevents.NewSourceClientMock(e.Services.Resources())
 	} else {
-		cloudEventsSourceOptions := mqttoptions.NewSourceOptions(e.Config.MessageBroker.MQTTOptions, e.Config.MessageBroker.ClientID, e.Config.MessageBroker.SourceID)
-		e.Clients.CloudEventsSource, err = cloudevents.NewSourceClient(cloudEventsSourceOptions, e.Services.Resources())
-		if err != nil {
-			glog.Errorf("Unable to create CloudEvents Source client: %s", err.Error())
-			return err
+		var cloudEventsSourceOptions *options.CloudEventsSourceOptions
+		if e.Config.MessageBroker.MessageBrokerType == work.ConfigTypeMQTT {
+			mqttOptions, err := mqtt.BuildMQTTOptionsFromFlags(e.Config.MessageBroker.MessageBrokerConfig)
+			if err != nil {
+				glog.Errorf("Unable to build MQTT options: %s", err.Error())
+				return err
+			}
+			cloudEventsSourceOptions = mqtt.NewSourceOptions(mqttOptions, e.Config.MessageBroker.ClientID, e.Config.MessageBroker.SourceID)
+		} else if e.Config.MessageBroker.MessageBrokerType == work.ConfigTypeKafka {
```

Comment:
instead of if/else , can this be an injectable implementation of an interface?

Reply:
Hi @markturansky I added a TODO to create a CloudEventsSourceOptionsBuilder to build the CloudEventsSourceOptions in sdk repo. Thanks.

Reply:
Done